### DISCUSSION TITLE
Add 25ppm when 'no route' n times on rebalance

### DISCUSSION THREAD

**Discussion by scissorstail** [cryptosharks131/lndg#203] [Ideas]
1. The variable 'n' has a value that can range from 1 to 1000.
2. If the condition "no route" occurs more than 'n' times during a rebalance, trigger 25ppm adjustment.
3. After the "UpdateHours" have passed, 25ppm is added and counting is reset to 0.
4. This process is repeated until the updated "ppm" value is lower than "MaxRate".

I believe this process helps to keep the rebalance running smoothly.

**Comment by cryptosharks131** [comment]
The issue I see with `No Route` is that is has no indication of any demand for the channel. For example, if a peer has little inbound capacity of their own, most of your rebalances will `No Route`, increasing fees and potentially buying some liquidity that could be difficult selling later at that price.

**Comment by scissorstail** [comment]
Hmmm... So is it possible to make **the autopilot automatically exclude that channel if it gets 'no_route' more than 'n' times**, and **then make it impossible for the autopilot to register again for a certain amount of time** (between 1 and 24 hours). Multiple 'no_route' channels interfere with rebalancing. It would be nice to rebalance other channels instead of constantly displaying meaningless 'no_route'.

And there is one more similar issue. **Channels that are 'disabled' by the remote cannot be rebalanced**. It would be nice if the autopilot **do not add 'disabled' channels** and automatically **exclude them if they become 'disabled'** during rebalance.

thanks.

`Translated with www.DeepL.com/Translator (free version)`

**Comment by cryptosharks131** [comment]
There is no real harm to `No Route` I am aware of. LND will try every path it can find within your time window, if it runs out of time, you get `Timeout` but if it runs out of paths you get `No Route`. If there are no routes to try, the rebalancer moves on right away and no time is wasted.
